.circleci/reportgenerator/main.go

```diff
@@ -0,0 +1,292 @@
+this issue is open, will be added as comments with more information to this issue.
+`
+	issueCommentTemplate = `
+Link to latest failed build: ${linkToBuild}
+
+${failedTests}
+`
+)
+
+func (rg reportGenerator) templateHelper(param string) string {
+	switch param {
+	case "jobName":
+		return "`" + rg.envVariables[jobNameKey] + "`"
+	case "linkToBuild":
+		return os.Getenv(circleBuildURLKey)
+	case "failedTests":
+		return rg.getFailedTests()
+	default:
+		return ""
+	}
+}
+
+// getExistingIssues gathers an existing GitHub Issue related to previous failures
+// of the same job.
+func (rg *reportGenerator) getExistingIssue() *github.Issue {
+	issues, response, err := rg.client.Issues.ListByRepo(
+		rg.ctx,
+		rg.envVariables[projectUsernameKey],
+		rg.envVariables[projectRepoNameKey],
+		&github.IssueListByRepoOptions{
+			State:  "open",
+			Labels: []string{ciFailureGithubLabel},
+		},
+	)
+	if err != nil {
+		rg.logger.Fatal("Failed to search GitHub Issues", zap.Error(err))
+	}
+
+	if response.StatusCode != http.StatusOK {
+		body, _ := ioutil.ReadAll(response.Body)
+		rg.logger.Fatal(
+			"Unexpected response from GitHub",
+			zap.String("status_code", string(response.StatusCode)),
+			zap.String("response", string(body)),
+			zap.String("url", response.Request.URL.String()),
+		)
+	}
+
+	requiredTitle := rg.getIssueTitle()
+	for _, issue := range issues {
+		if *issue.Title == requiredTitle {
+			return issue
+		}
+	}
+
+	return nil
+}
+
+// commentOnIssue adds a new comment on an existing GitHub issue with
+// information about the latest failure. This method is expected to be
+// called only if there's an existing open Issue for the current job.
+func (rg *reportGenerator) commentOnIssue(issue *github.Issue) *github.IssueComment {
+	body := os.Expand(issueCommentTemplate, rg.templateHelper)
+
+	issueComment, response, err := rg.client.Issues.CreateComment(
+		rg.ctx,
+		rg.envVariables[projectUsernameKey],
+		rg.envVariables[projectRepoNameKey],
+		*issue.Number,
+		&github.IssueComment{
+			Body: &body,
+		},
+	)
+	if err != nil {
+		rg.logger.Fatal("Failed to search GitHub Issues", zap.Error(err))
+	}
+
+	if response.StatusCode != http.StatusCreated {
+		body, _ := ioutil.ReadAll(response.Body)
+		rg.logger.Fatal(
+			"Unexpected response from GitHub",
+			zap.String("status_code", string(response.StatusCode)),
+			zap.String("response", string(body)),
+			zap.String("url", response.Request.URL.String()),
+		)
+	}
+
+	return issueComment
+}
+
+// createIssue creates a new GitHub Issue corresponding to a build failure.
+func (rg *reportGenerator) createIssue() *github.Issue {
+	title := rg.getIssueTitle()
+	body := os.Expand(issueBodyTemplate, rg.templateHelper)
+
+	issue, response, err := rg.client.Issues.Create(
+		rg.ctx,
+		rg.envVariables[projectUsernameKey],
+		rg.envVariables[projectRepoNameKey],
+		&github.IssueRequest{
+			Title:  &title,
+			Body:   &body,
+			Labels: &[]string{ciFailureGithubLabel},
```

Comment:
Since this job will be using the same token @owais is referring to here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/278#discussion_r436086112, I believe we won't be able to add labels or set assignees at the moment. Will update to not set labels to begin with.